src/Platforms/SQLitePlatform.php

```diff
@@ -917,7 +933,11 @@ private function getForeignKeysInAlteredTable(TableDiff $diff): array
                         ->getValue(),
                 );
 
-                $foreignKeys[$constraintKey] = $constraint;
+                if (isset($keysByName[$constraintKey])) {
+                    $foreignKeys[$keysByName[$constraintKey]] = $constraint;
```

Comment:
This looks like an error state to me. If a constraint with the same name already exists, it shouldn't be silently overwritten.

Let's make an it assertion for now that `$keysByName[$constraintKey]` is not set and revisit later.

Also, the new name and key need to be added to `$keysByName` to prevent the overwrite by the subsequent ones.